components/TextInput.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Controller, Control } from 'react-hook-form';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { FormValues } from '@/types';
+
+type TextInputProps = TextFieldProps &  {
+  name: keyof FormValues;
+  label: string;
+  control: Control<FormValues>;
+  rules?: object;
+  fullWidth?: boolean;
+  halfWidth?: boolean;
+}
+
+const TextInput: React.FC<TextInputProps> = ({ name, label, control, rules, fullWidth, halfWidth }) => {
```

Comment:
If you have many props on a single line, break them onto separate lines for ease of readability.